src/app/meta-pixel.service.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable } from '@angular/core';
+import { NavigationEnd, Router } from '@angular/router';
+
+import { environment } from '../environments/environment';
+
+declare const fbq: (action: string, eventOrId: string, params?: object) => void;
```

Comment:
To be explicit about the fact that fbq might not be initialised if the stuff from meta doesn't load. This means the TS compiler will force us to check that it's a function before we call it.
```suggestion
declare const fbq: ((action: string, eventOrId: string, params?: object) => void) | undefined;
```